ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +1052,242 @@ def test_server_option_with_unreachable_ad(self):
             self.master.run_command(['rm', '-f', ad_zone_file])
             tasks.configure_dns_for_trust(self.master, self.ad)
             self.remove_trust(self.ad)
+
+
+class TestNonPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+
+    @classmethod
+    def install(cls, mh):
+        if not cls.master.transport.file_exists('/usr/bin/rpcclient'):
+            raise pytest.skip("Package samba-client not available "
+                              "on {}".format(cls.master.hostname))
+        super(TestNonPosixAutoPrivateGroup, cls).install(mh)
+        cls.ad = cls.ads[0]
+        cls.ad_domain = cls.ad.domain.name
+        tasks.install_adtrust(cls.master)
+        tasks.sync_time(cls.master, cls.ad)
+        cls.existing_rangename = cls.ad_domain.upper() + "_id_range"
+
+    def test_add_nonposix_trust(self):
```

Comment:
You can factor code and define the following in BaseTestTrust:
```
def add_trust(self, rangetype):
    tasks.configure_dns_for_trust(self.master, self.ad)
    tasks.establish_trust_with_ad(self.master, self.ad_domain, extra_args=['--range-type', rangetype])
```
Then in this class:
```
def test_add_nonposix_trust(self):
    self.add_trust('ipa-ad-trust')
```
and in TestPosixAUtoPrivateGroup:
```
def test_add_posix_trust(self):
    self.add_trust('ipa-ad-trust-posix')
```